libs/api/domains/application/src/lib/application.module.ts

```diff
@@ -1,23 +1,32 @@
-import { Module } from '@nestjs/common'
+import { DynamicModule } from '@nestjs/common'
 import fetch from 'isomorphic-fetch'
 import { ApplicationResolver } from './application.resolver'
 import { ApplicationService } from './application.service'
 import { ApplicationsApi, Configuration } from '../../gen/fetch'
 
-@Module({
-  providers: [
-    ApplicationResolver,
-    ApplicationService,
-    {
-      provide: ApplicationsApi,
-      useFactory: () =>
-        new ApplicationsApi(
-          new Configuration({
-            fetchApi: fetch,
-            basePath: 'http://localhost:3333',
-          }),
-        ),
-    },
-  ],
-})
-export class ApplicationModule {}
+export interface Config {
+  baseApiUrl: string
+}
+
+export class ApplicationModule {
```

Comment:
Don't you wrap it with the Module decorator?

Reply:
Consider it wrapped